http/src/test/scala/zipkin/finagle/http/HttpRawZipkinTracerTest.scala

```diff
@@ -0,0 +1,102 @@
+package zipkin.finagle.http
+
+import java.net.URI
+import java.util
+import java.util.Arrays.asList
+
+import com.twitter.finagle.ChannelClosedException
+import com.twitter.finagle.stats.InMemoryStatsReceiver
+import com.twitter.finagle.tracing.{Flags, SpanId, TraceId}
+import com.twitter.finagle.zipkin.core.{BinaryAnnotation, Endpoint, Span, ZipkinAnnotation}
+import com.twitter.util.{Await, Base64StringEncoder, Time}
+import org.junit.{ClassRule, Test}
+import org.scalatest.junit.JUnitSuite
+import zipkin.Codec
+import zipkin.junit.{HttpFailure, ZipkinRule}
+
+object HttpRawZipkinTracerTest {
+  // Singleton as the test needs to read the actual port in use
+  val zipkinRule = new ZipkinRule()
+
+  // Scala cannot generate fields with public visibility, so use a def instead.
+  @ClassRule def zipkinDef = zipkinRule
+}
+
+class HttpRawZipkinTracerTest extends JUnitSuite {
+
+  import HttpRawZipkinTracerTest.zipkinRule
+
+  val statsReceiver = new InMemoryStatsReceiver
+
+  val traceId = TraceId(Some(SpanId(123)), Some(SpanId(123)), SpanId(123), None, Flags().setDebug)
+
+  @Test def sendSpans {
+    val localEndpoint = Endpoint(2323, 23)
+    val remoteEndpoint = Endpoint(333, 22)
+
+    val annotations = Seq(
+      ZipkinAnnotation(Time.fromSeconds(123), "cs", localEndpoint),
+      ZipkinAnnotation(Time.fromSeconds(126), "cr", localEndpoint),
+      ZipkinAnnotation(Time.fromSeconds(123), "ss", remoteEndpoint),
```

Comment:
note this is unrealistic as a tracer doesn't report spans outside its process. it is copy/pasted from the other test.